packages/nimble/R/cppInterfaces_nimbleFunctions.R

```diff
@@ -899,88 +892,58 @@ copyFromRobject <- function(Robj, cppNames, cppCopyTypes, basePtr, symTab, dll)
-            next
-        }
-        else if(cppCopyTypes[[v]] == "modelValuesPtr"){
+        },
+        "modelValuesPtr" = {
             curObj <- Robj[[v]]
             mvPtr = curObj$modelValues$CobjectInterface$componentExtptrs[[curObj$var]]
             getSetModelValuesPtr(v, mvPtr, basePtr, dll = dll)
-            ## .self[[v]] <<- mvPtr
-            next
-        }
-        else if(cppCopyTypes[[v]] == 'numericList'){
+        },
+        'numericList' = {
             stop('numericList is not working\n')
-            ## rawPtr = .Call('getModelObjectPtr', .self$.basePtr, v)
-            ## .self[[v]] <<- numericList(buildType = 'C', extPtr = rawPtr)
-            ## nRows = Robj[[v]]$nRow
-            ## resize(.self[[v]], nRows)
-            ## for(i in 1:nRows){
-            ##     copyDims = max(c(1, dimOrLength[[v]][[i]]) )
-            ##     d1 = copyDims[1]
-            ##     d2 = copyDims[2]
-            ##     d3 = copyDims[3]
-            ##     setSize(.self[[v]], row = i, d1, d2, d3)
-            ##     .self[[v]][[i]] <<- Robj[[v]][[i]]
-            ## }
-            next
-        }
-        else if(cppCopyTypes[[v]] == 'indexedNodeInfoTable') {
+        },
+        'indexedNodeInfoTable' = {
             populateIndexedNodeInfoTable(fxnPtr = basePtr, Robject = Robj, indexedNodeInfoTableName = v, dll = dll)
-        }
-        else if(cppCopyTypes[[v]] == 'characterVector') {
+        },
+        'characterVector' = {
             getSetCharacterVector(v, Robj[[v]], basePtr, dll = dll)
-            ##.self[[v]] <<- Robj[[v]]
-            next
-        }
-        else if(cppCopyTypes[[v]] == 'characterScalar') {
+        },
+        'characterScalar' = {
             getSetCharacterScalar(v, Robj[[v]], basePtr, dll = dll)
-            next
-        }
-        else if(cppCopyTypes[[v]] == 'numericVector') {
-            ##message('copying numericVector (copyFromRobject)')
+        },
+        'numericVector' = {
             getSetNumericVector(v, Robj[[v]], basePtr, dll = dll)
-            ##.self[[v]] <<- Robj[[v]]
-            next
-        }
-        else if(cppCopyTypes[[v]] == 'doubleScalar') {
+        },
+        'doubleScalar' = {
             getSetDoubleScalar(v, Robj[[v]], basePtr, dll = dll)
-            next
-        }
-        else if(cppCopyTypes[[v]] == 'integerScalar') {
+        },
+        'integerScalar' = {
             getSetIntegerScalar(v, Robj[[v]], basePtr, dll = dll)
-            next
-        }
-        else if(cppCopyTypes[[v]] == 'logicalScalar') {
+        },
+        'logicalScalar' = {
             getSetLogicalScalar(v, Robj[[v]], basePtr, dll = dll)
-            next
-        }
-        else if(!(cppCopyTypes[[v]] %in% c('copierVector'))) {
-            warning(paste0("Note: cppCopyTypes not recognized. Type = ", cppCopyTypes[[v]], "\n"), call. = FALSE)
+        },
+        { ## default:
+            if(!(cppCopyTypes[[v]] %in% c('copierVector'))) {
```

Comment:
Consider breaking `copierVector` out into another case, and maybe moving this case to the top?
```
},
'copierVector': {},
{  # default
  warning(paste0("Note: cppCopyTypes not recognized. Type = ", cppCopyTypes[[v]], "\n"), call. = FALSE)
}
)
```